tests/FunctionalTestCase.php

```diff
@@ -53,23 +54,32 @@ protected function getFeatureCompatibilityVersion(ReadPreference $readPreference
             return $this->getServerVersion($readPreference);
         }
 
-        $cursor = $this->manager->executeCommand(
-            'admin',
-            new Command(['getParameter' => 1, 'featureCompatibilityVersion' => 1]),
-            $readPreference ?: new ReadPreference(ReadPreference::RP_PRIMARY)
-        );
-
-        $cursor->setTypeMap(['root' => 'array', 'document' => 'array']);
-        $document = current($cursor->toArray());
-
-        // MongoDB 3.6: featureCompatibilityVersion is an embedded document
-        if (isset($document['featureCompatibilityVersion']['version']) && is_string($document['featureCompatibilityVersion']['version'])) {
-            return $document['featureCompatibilityVersion']['version'];
-        }
-
-        // MongoDB 3.4: featureCompatibilityVersion is a string
-        if (isset($document['featureCompatibilityVersion']) && is_string($document['featureCompatibilityVersion'])) {
-            return $document['featureCompatibilityVersion'];
+        try {
+            $cursor = $this->manager->executeCommand(
+                'admin',
+                new Command(['getParameter' => 1, 'featureCompatibilityVersion' => 1]),
+                $readPreference ?: new ReadPreference(ReadPreference::RP_PRIMARY)
+            );
+
+            $cursor->setTypeMap(['root' => 'array', 'document' => 'array']);
+            $document = current($cursor->toArray());
+
+            // MongoDB 3.6: featureCompatibilityVersion is an embedded document
+            if (isset($document['featureCompatibilityVersion']['version']) && is_string($document['featureCompatibilityVersion']['version'])) {
+                return $document['featureCompatibilityVersion']['version'];
+            }
+
+            // MongoDB 3.4: featureCompatibilityVersion is a string
+            if (isset($document['featureCompatibilityVersion']) && is_string($document['featureCompatibilityVersion'])) {
+                return $document['featureCompatibilityVersion'];
+            }
+        } catch (CommandException $e) {
+            /* It is not possible to get the featureCompatibilityVersion from mongos, so if the
```

Comment:
Did you write this with the expectation that a future version of mongos might support querying for FCV?

Would it make sense to leave most of the original method as-is and introduce an early conditional that instead selects a server and defers to `$this->getServerVersion($readPreference)` if its type is a mongos node?

That would also avoid using a magic constant (`17`) below, and allow us to get rid of the entire try/catch here.